### PR TITLE
Anko - Update group data source

### PR DESCRIPTION
**Pull request by at-cuongcao**
#### Related issues (Required)
- Task #171 

#### Description (Required)
- [x] Update group data source

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt [hunk 1]
```diff
@@ -264,6 +267,93 @@ class GroupRemoteDataSource : GroupDataSource {
...
+                                        .addOnFailureListener {
+                                            result.onError(it)
+                                        }
+                            }
+                        }
+                    })
+                }
+                .addOnFailureListener {
+                    result.onError(it)
+                }
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt [hunk 2]
```diff
@@ -264,6 +267,93 @@ class GroupRemoteDataSource : GroupDataSource {
         return HypertrackApi.instance.getGroupMembers(groupId).map { it.results }
     }
 
+    override fun acceptInvite(userId: String, invite: Invite): Single<Boolean> {
+        val gson = Gson()
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/group/GroupActivity.kt [hunk 3]
```diff
@@ -17,6 +20,19 @@ class GroupActivity : BaseActivity() {
         super.onCreate(savedInstanceState)
         ui = GroupActivityUI()
         ui.setContentView(this)
+
+        addDisposables(
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupActivity.kt]: please remove dummy data

File: app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt [hunk 4]
```diff
@@ -264,6 +267,92 @@ class GroupRemoteDataSource : GroupDataSource {
         return HypertrackApi.instance.getGroupMembers(groupId).map { it.results }
     }
 
+    override fun acceptInvite(userId: String, invite: Invite): Single<Boolean> {
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: Split to some private functions

File: app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt [hunk 5]
```diff
@@ -264,6 +267,102 @@ class GroupRemoteDataSource : GroupDataSource {
...
+                override fun onDataChange(p0: DataSnapshot?) {
+                    if (p0?.value == null) {
+                        acceptInviteWhenUserDoNotHaveRequestAtTime(result, userId, invite)
+                    } else {
+                        acceptInviteWhenUserHaveAnotherRequestAtTime(result, userId, invite, p0.value.toString())
+                    }
+                }
+            })
+        }.addOnFailureListener {
+            result.onError(it)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt [hunk 6]
```diff
@@ -264,6 +267,102 @@ class GroupRemoteDataSource : GroupDataSource {
...
+                    "/request/${invite.to}")
+            invite.to = userId
+            requestRef.setValue(invite).addOnSuccessListener {
+                result.onSuccess(true)
+            }.addOnFailureListener {
+                result.onError(it)
+            }
+        }
+    }
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: WARNING: Unexpected indentation

**Pull request merged**